src/cider/nrepl/middleware/out.clj

```diff
@@ -0,0 +1,95 @@
+(ns cider.nrepl.middleware.out
+  "Change *out* to print on sessions in addition to process out.
+
+  Automatically changes the root binding of *out* to print to any
+  active sessions. An active session is one that has sent at least one
+  \"eval\" op.
+
+  We use an eval message, instead of the clone op, because there's no
+  guarantee that the channel that sent the clone message will properly
+  handle output replies."
+  (:require [cider.nrepl.middleware.util.cljs :as cljs]
+            [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
+            [clojure.tools.nrepl.middleware.interruptible-eval :as ie]
+            [clojure.tools.nrepl.middleware.session :as session])
+  (:import [java.io PrintWriter Writer]))
+
+;;; OutStream
+(defonce original-out *out*)
+
+(declare tracked-sessions-map)
+
+(defmacro with-out-binding
+  "Run body with v bound to the output stream of each msg in msg-seq.
+  Also run body with v bound to `original-out`."
+  [[v msg-seq] & body]
+  `(do (let [~(with-meta v {:tag Writer}) original-out]
+         ~@body)
+       (doseq [{:keys [~'session] :as ~'msg} ~msg-seq]
+         (let [~(with-meta v {:tag Writer}) (get @~'session #'*out*)]
+           (try (binding [ie/*msg* ~'msg]
+                  ~@body)
+                ;; If a channel is faulty, dissoc it.
+                (catch Exception ~'e
+                  (swap! tracked-sessions-map dissoc
+                         (:id (meta ~'session)))))))))
+
+(defn fork-out
+  "Returns a PrintWriter suitable for binding as *out* or *err*. All
+  operations are forwarded to all output bindings in the sessions of
+  messages in addition to the server's usual PrintWriter (saved in
+  `original-out`)."
+  [messages]
+  (PrintWriter. (proxy [Writer] []
+                  (close [] (.flush ^Writer this))
+                  (write
+                    ([x]
+                     (with-out-binding [out messages]
+                       (.write out x)))
+                    ([x ^Integer off & [^Integer len]]
```

Comment:
I still don't get why you need this optional argument, there's only arity 1 and 3 .write methods in http://docs.oracle.com/javase/7/docs/api/java/io/Writer.html, so len will never not be null if it gets called.